### DISCUSSION TITLE
tp.file.rename doesn't use the rest of the template's commands

### DISCUSSION THREAD

**Discussion by evanharmon1** [SilentVoid13/Templater#696] [Help]
When I use `<% await tp.file.rename("new file name") %>` the template renames correctly but it ignores the rest of the template commands I want to use. I have tried a number of different things like reordering the template commands, trying with and without await, etc. Below is the template file I am trying to use. And I am invoking it with the Templater: Create New Note from Template palette command with selected text. The resulting file is just a blank file with no yaml front matter and no content other than the note title at the top. Am I doing something wrong?

```
---
type: source
tags: source
---
# template$Source
<% await tp.file.rename("new file name") %>

<% await tp.file.selection() %>

<% await tp.file.selection().split('\n')[0]%>
```

**Comment by welpdx** [comment]
unable to replicate

<details>
           <summary style="font-size:20">Demo</summary>
           <img src="https://i.imgur.com/MvPni3V.gif" alt="demo" />
</details>

Note: 
- According to [src code](https://github.com/SilentVoid13/Templater/blob/097c7c0c99dc9f6932af6d44df9ebccfd995c4b3/src/core/functions/internal_functions/file/InternalModuleFile.ts#L296), `tp.file.selection()` is not an async function. There is no need to add `await` in front of it

**Reply by evanharmon1** [reply]
Thanks. Your demo is exactly what I would expect to happen and what I'm going for. I've tried with and without async like you pointed out. What exactly are the steps that you are taking after you highlight text? I have been using the Templater: Create new note from template command from the command palette (and with a keyboard shortcut). It is still only creating a new note and doesn't have any content other than the heading with the specified filename.

**Reply by evanharmon1** [reply]
I tested on a new from scratch vault and got the intended behavior. I found that it is due to a plugin I use called Filename Heading Sync. Disabling this fixed the issue. But that plugin is super useful to me so I'm not sure I want to disable it...

**Reply by evanharmon1** [reply]
Found a similar issue on that plugin's repo: https://github.com/dvcrn/obsidian-filename-heading-sync/issues/55

**Reply by welpdx** [reply]
For my demo, I used `Alt + E Templater: Open Insert Template modal`, but regardless, it seems like you have found your problem. 

I've never used the plugin you mentioned but after a quick look, I think a good workaround is to simply put your new name as the first header. For example, 
```js
---
type: source
tags: source
---
# replyback

<%  tp.file.selection() %>

<%  tp.file.selection().split('\n')[0]%>
```

<details>
           <summary style="font-size:20">Demo</summary>
           <img src="https://i.imgur.com/FQXjkDh.gif" alt="demo" />
</details>

You can also disable `Use file open hook` from the Filename Heading Sync plugin to make `tp.file.rename()` work but I guess that basically ruins the use case of that plugin.
